x-pack/plugins/lens/public/editor_frame_service/editor_frame/workspace_panel/workspace_panel.tsx

```diff
@@ -189,6 +195,23 @@ export function InnerWorkspacePanel({
     }
   }, [expression, localState.expressionBuildError]);
 
+  if (visualizeTriggerFieldContext && !localState.visualizeTriggerSuggestionsLoaded && !title) {
```

Comment:
We didn't do a good job so far and logic and UI is pretty intertwined right now - let's try to get better. This logic doesn't seem related to the workspace panel, can we move it into the suggestions helper collection and call it from the editor frame top level component? That way it's encapsulated a little better. Also it seems like in the editor frame you don't need a separate flag for remembering whether the suggestion is already applied because you can use a `useEffect` hook on the `allLoaded` variable (which means all datasources are ready).